storage/database/dynamodb.go

```diff
@@ -0,0 +1,484 @@
+// This file is part of the klaytn library.
+//
+// The klaytn library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The klaytn library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the klaytn library. If not, see <http://www.gnu.org/licenses/>.
+
+package database
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/rcrowley/go-metrics"
+
+	"github.com/pkg/errors"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbattribute"
+	"github.com/klaytn/klaytn/log"
+)
+
+var dataNotFoundErr = errors.New("data is not found with the given key")
+var nilDynamoConfigErr = errors.New("attempt to create DynamoDB with nil configuration")
+
+const dynamoWriteSizeLimit = 399 * 1024 // The maximum write size is 400KB including attribute names and values
+const dynamoBatchSize = 25
+const dynamoMaxRetry = 5
+
+// dynamo table provisioned setting
+const dynamoReadCapacityUnits = 10000
+const dynamoWriteCapacityUnits = 10000
+
+// batch write
+const WorkerNum = 10
+const itemChanSize = WorkerNum * 2
+
+var overSizedDataPrefix = []byte("oversizeditem")
+
+type DynamoDBConfig struct {
+	Region             string
+	Endpoint           string
+	TableName          string
+	ReadCapacityUnits  int64
+	WriteCapacityUnits int64
+}
+
+type batchWriteWorkerInput struct {
+	items []*dynamodb.WriteRequest
+	wg    *sync.WaitGroup
+}
+
+type dynamoDB struct {
+	config *DynamoDBConfig
+	db     *dynamodb.DynamoDB
+	fdb    fileDB
+	logger log.Logger // Contextual logger tracking the database path
+
+	// worker pool
+	quitCh  chan struct{}
+	writeCh chan *batchWriteWorkerInput
+
+	batchWriteTimeMeter       metrics.Meter
+	batchWriteCountMeter      metrics.Meter
+	batchWriteSizeMeter       metrics.Meter
+	batchWriteSecPerItemMeter metrics.Meter
+	batchWriteSecPerByteMeter metrics.Meter
+}
+
+type DynamoData struct {
+	Key []byte `json:"Key" dynamodbav:"Key"`
+	Val []byte `json:"Val" dynamodbav:"Val"`
+}
+
+// TODO-Klaytn: remove the test config when flag setting is completed
+/*
+ * Please Run DynamoDB local with docker
+ * $ docker pull amazon/dynamodb-local
+ * $ docker run -d -p 8000:8000 amazon/dynamodb-local
+ */
+func createTestDynamoDBConfig() *DynamoDBConfig {
+	return &DynamoDBConfig{
+		Region:             "ap-northeast-2",
+		Endpoint:           "https://dynamodb.ap-northeast-2.amazonaws.com",
+		TableName:          "dynamo-test-tmp",
```

Comment:
Is there a reason why there is no timestamp in the table name? I am worried if more than one people run this code without the table name configuration.

Reply:
When `go test` runs, `init` function in "db_manager_test.go" generates a dynamoDB table and doesn't delete the table after the execution. If we added a timestamp, many unused tables can be generated. 
And `createTestDynamoDBConfig` should be eliminated soon as I commented above.